exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -18,109 +18,147 @@ import (
-			startErr = errors.New("cannot find log group and stream")
-			return
-		}
-		stream := out.LogStreams[0]
-		if stream.UploadSequenceToken == nil {
-			e.logger.Debug("CloudWatch sequence token is nil, will assume empty")
-			return
-		}
-		e.seqToken = *stream.UploadSequenceToken
-	})
-	return startErr
-}
+	// create CWLogs client with aws session config
+	svcStructuredLog := cwlogs.NewClient(params.Logger, awsConfig, params.BuildInfo, expConfig.LogGroupName, session)
+	collectorIdentifier, err := uuid.NewRandom()
+
+	if err != nil {
+		return nil, err
+	}
+
+	expConfig.Validate()
+
+	logsExporter := &exporter{
+		svcStructuredLog: svcStructuredLog,
+		config:           config,
+		logger:           params.Logger,
+		retryCount:       *awsConfig.MaxRetries,
+		collectorID:      collectorIdentifier.String(),
+	}
+	logsExporter.groupStreamToPusherMap = map[string]map[string]cwlogs.Pusher{}
+
+	return exporterhelper.NewLogsExporter(
+		config,
+		params,
+		logsExporter.PushLogs,
+		exporterhelper.WithQueue(expConfig.enforcedQueueSettings()),
+		exporterhelper.WithRetry(expConfig.RetrySettings),
+	)
 
-func (e *exporter) Shutdown(ctx context.Context) error {
-	// TODO(jbd): Signal shutdown to flush the logs.
-	return nil
 }
 
-func (e *exporter) PushLogs(ctx context.Context, ld pdata.Logs) (err error) {
+func (e *exporter) PushLogs(ctx context.Context, ld pdata.Logs) error {
 	// TODO(jbd): Relax this once CW Logs support ingest
 	// without sequence tokens.
 	e.seqTokenMu.Lock()
 	defer e.seqTokenMu.Unlock()
 
+	exp := e.config.(*Config)
+	cwLogsPusher := e.getLogPusher(exp.LogGroupName, exp.LogStreamName)
 	logEvents, _ := logsToCWLogs(e.logger, ld)
 	if len(logEvents) == 0 {
 		return nil
 	}
 
-	e.logger.Debug("Putting log events", zap.Int("num_of_events", len(logEvents)))
-	input := &cloudwatchlogs.PutLogEventsInput{
-		LogGroupName:  aws.String(e.config.LogGroupName),
-		LogStreamName: aws.String(e.config.LogStreamName),
-		LogEvents:     logEvents,
+	e.logger.Info("Putting log events", zap.Int("num_of_events", len(logEvents)))
+
+	for _, logEvent := range logEvents {
+		logEvent := &cwlogs.Event{
+			InputLogEvent: logEvent,
+			GeneratedTime: time.Now(),
+		}
+		e.logger.Debug("Adding log event", zap.Any("event", logEvent))
+		err := cwLogsPusher.AddLogEntry(logEvent)
+		if err != nil {
+			e.logger.Error("Failed ", zap.Int("num_of_events", len(logEvents)))
+		}
 	}
-	if e.seqToken != "" {
-		input.SequenceToken = aws.String(e.seqToken)
-	} else {
-		e.logger.Debug("Putting log events without a sequence token")
+	e.logger.Debug("Log events are successfully put")
+	flushErr := cwLogsPusher.ForceFlush()
+	if flushErr != nil {
+		e.logger.Error("Error force flushing logs. Skipping to next logPusher.", zap.Error(flushErr))
+		return flushErr
 	}
+	return nil
+}
 
-	out, err := e.client.PutLogEvents(input)
-	if err != nil {
-		return err
+func (e *exporter) ConsumeLogs(ctx context.Context, md pdata.Logs) error {
+	return e.PushLogs(ctx, md)
```

Comment:
It doesn't seem that `PushLogs()` is needed anymore.  I would inline its implementation here as this is where someone familiar with the collector will start looking.

Reply:
removed Pushlogs and moved its code to ConsumeLogs